smac/utils/io/traj_logging.py

```diff
@@ -146,3 +153,66 @@ def _add_in_aclib_format(self, train_perf, incumbent_id, incumbent,
         with open(self.aclib_traj_fn, "a") as fp:
             json.dump(traj_entry, fp)
             fp.write("\n")
+
+    @staticmethod
+    def read_traj_aclib_format(fn: str, cs: ConfigurationSpace):
+        """
+            reads trajectory from file
+
+            Parameters
+            ----------
+            fn: str
+                filename with saved runhistory in self._add_in_aclib_format format
+            cs: ConfigurationSpace
+                Configuration Space to translate dict object into Confiuration object
+
+            Returns
+            -------
+            trajectory: list
+                each entry in the list is a dictionary of the form
+                {"cpu_time": float,
+                 "total_cpu_time": None, # TODO
+                 "wallclock_time": float,
+                 "evaluations": int
+                 "cost": float,
+                 "incumbent": Configuration
+        """
+
+        trajectory = []
+        with open(fn) as fp:
+            for line in fp:
+                entry = json.loads(line)
+                entry["incumbent"] = TrajLogger._convert_dict_to_config(
+                    entry["incumbent"], cs=cs)
+                trajectory.append(entry)
+
+        return trajectory
+
+    @staticmethod
+    def _convert_dict_to_config(config_list: typing.List[str], cs: ConfigurationSpace):
+        '''
+            since we save a configurations in a dictionary str->str
+            we have to try to figure out the type (int, float, str) of each parameter value
+
+            Parameters
+            ----------
+            config_list: typing.List[str]
+                configuration as a list of "str='str'"
+            cs: ConfigurationSpace
+                Configuration Space to translate dict object into Confiuration object
+        '''
+        config_dict = {}
+        for param in config_list:
+            k,v = param.split("=")
+            v = v.strip("'")
+            hp = cs.get_hyperparameter(k)
+            if isinstance(hp, FloatHyperparameter):
```

Comment:
It would be great if SMAC does not need to implement the logic to convert types of parameter values, but the config space could fix all types since it is more aware of all its hyperparameter types. 
This would also prevent that we will miss to update this code in the future when we would add a new hyperparameter type in the configspace.